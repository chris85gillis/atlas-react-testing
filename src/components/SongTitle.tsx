import React from 'react';

type SongTitleProps = {
  title: string;
  author: string;
};

const SongTitle: React.FC<SongTitleProps> = ({ title, author }) => {
  return (
    <div className="text-center mt-4">
      <h1 className="text-2xl font-bold text-nightText">{title}</h1>
      <p className="text-lg text-nightSecondary">{author}</p>
    </div>
  );
};

export default SongTitle;