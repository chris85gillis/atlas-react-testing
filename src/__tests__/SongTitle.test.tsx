import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';

describe('SongTitle', () => {
  it('renders correctly with title and author', () => {
    const { asFragment } = render(<SongTitle title="Song Title" author="Artist Name" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with missing author', () => {
    const { asFragment } = render(<SongTitle title="Song Title" author="" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with missing title', () => {
    const { asFragment } = render(<SongTitle title="" author="Artist Name" />);
    expect(asFragment()).toMatchSnapshot();
  });
});