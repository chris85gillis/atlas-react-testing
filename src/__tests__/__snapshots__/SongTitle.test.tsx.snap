// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > renders correctly with missing author 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-4"
  >
    <h1
      class="text-2xl font-bold text-nightText"
    >
      Song Title
    </h1>
    <p
      class="text-lg text-nightSecondary"
    />
  </div>
</DocumentFragment>
`;

exports[`SongTitle > renders correctly with missing title 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-4"
  >
    <h1
      class="text-2xl font-bold text-nightText"
    />
    <p
      class="text-lg text-nightSecondary"
    >
      Artist Name
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle > renders correctly with title and author 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-4"
  >
    <h1
      class="text-2xl font-bold text-nightText"
    >
      Song Title
    </h1>
    <p
      class="text-lg text-nightSecondary"
    >
      Artist Name
    </p>
  </div>
</DocumentFragment>
`;
